geo/src/algorithm/relate/geomgraph/intersection_matrix.rs

```diff
@@ -268,6 +268,161 @@ impl IntersectionMatrix {
+    ///
+    /// let intersection = line_string.relate(&poly);
+    /// assert_eq!(intersection.is_crosses(), true);
+    /// ```
+    /// # If any of the following do not hold, the function will return `false`
+    /// 1. The intersection of the interiors of the geometries must be non-empty
+    /// 2. The intersection must have dimension less than the maximum dimension of the two input geometries
+    /// 3. The intersection of the two geometries must not equal either geometry
+    pub fn is_crosses(&self) -> bool {
+        if self.0[CoordPos::Inside][CoordPos::Outside]
+            != self.0[CoordPos::Outside][CoordPos::Inside]
+        {
+            // input geom dimensions differ: proceed to calculating relationships between dimensions
+            let ii = self.0[CoordPos::Inside][CoordPos::Inside];
+            let ie = self.0[CoordPos::Inside][CoordPos::Outside];
+            let ei = self.0[CoordPos::Outside][CoordPos::Inside];
+            let a_lessthan_b = ii == ie;
+            let a_greaterthan_b = ii == ei;
+            // something has gone horribly wrong
+            assert!(a_lessthan_b != a_greaterthan_b);
+            if a_lessthan_b {
+                // [T*T******]
+                return self.0[CoordPos::Inside][CoordPos::Inside] != Dimensions::Empty
+                    && self.0[CoordPos::Inside][CoordPos::Outside] != Dimensions::Empty;
+            }
+            if a_greaterthan_b {
+                // [T*****T**]
+                return self.0[CoordPos::Inside][CoordPos::Inside] != Dimensions::Empty
+                    && self.0[CoordPos::Outside][CoordPos::Inside] != Dimensions::Empty;
+            }
+        } else {
+            // dimensions are equal: this is only permitted for line / line
+            if self.0[CoordPos::Inside][CoordPos::Outside] == Dimensions::OneDimensional {
+                // line / line: [0********]
+                return self.0[CoordPos::Inside][CoordPos::Inside] == Dimensions::ZeroDimensional;
+            } else {
+                return false;
+            }
+        }
+        true
+    }
+
+    /// Returns `true` if geometry A and B "spatially overlap". Two geometries overlap if they have the
+    /// same dimension, their interiors intersect in that dimension. and each has at least one point
+    /// inside the other (or equivalently, neither one covers the other)
+    ///
+    /// ```
+    /// use geo_types::{Polygon, polygon};
+    /// use geo::relate::Relate;
+    ///
+    /// let poly1 = polygon![
+    ///     (x: 125., y: 179.),
+    ///     (x: 110., y: 150.),
+    ///     (x: 160., y: 160.),
+    ///     (x: 125., y: 179.),
+    /// ];
+    /// let poly2 = polygon![
+    ///     (x: 126., y: 179.),
+    ///     (x: 110., y: 150.),
+    ///     (x: 161., y: 160.),
+    ///     (x: 126., y: 179.),
+    /// ];
+    ///
+    /// let intersection = poly1.relate(&poly2);
+    /// assert_eq!(intersection.is_overlaps(), true);
+    /// ```
+    #[allow(clippy::nonminimal_bool)]
```

Comment:
Clippy has suggested simplified boolean logic but I want to maintain what's here as it's easy to check against the documented relations.